.gitlab-ci.yml

```diff
@@ -1,31 +1,49 @@
-before_script:
-  # check the availability of some commands:
-  - command >/dev/null -v python3 jq
-
-after_script:
-  - tools/docker/stop.sh -k -r
-
 build-in-docker:
   stage: build
+  image:
+    name: prplfoundationinc/prplmesh-builder:ubuntu-18.04
+    entrypoint: [""]
+  variables:
+    CMAKE_BUILD_TYPE: "Release"
   script:
-    - cd tools/docker
-    # Try to pull from dockerhub, build the container images if that fails
-    - ./image-pull.sh || ./image-build.sh
     # Run clang-format and check there are no modified files by checking git status
     - ./clang-format.sh
     - git diff-index --exit-code HEAD
-    - ./build.sh map -f CMAKE_BUILD_TYPE=Debug BUILD_TESTS=ON
+    - cmake -DBUILD_TESTS=ON -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DCMAKE_INSTALL_PREFIX=build/install -H. -Bbuild -G Ninja
+    - ninja -C build install
     # Check that the AutoGenerated files are correct by checking git status
     - git update-index -q --refresh
     - git diff-index --exit-code HEAD
+
+  artifacts:
+    paths:
+      - build
   tags:
-    - dockerbuild
+    - docker
+    - specific
 
 run-tests:
   stage: test
+  image: $CI_REGISTRY_IMAGE/tests-runner
   script:
     - cd tools/docker
-    - ./run.sh -f -d --entrypoint "$PWD/../../../build/install/bin/tests/tlvf_test"
+    # Try to pull from dockerhub, build the container images if that fails
+    - ./image-pull.sh || ./image-build.sh
+    - ./run.sh -f --entrypoint "$PWD/../../build/install/bin/tests/tlvf_test"
     - ./tests/test_flows.sh -v
+
+  before_script:
+    # start the docker daemon and give it some time to start:
+    - dockerd-entrypoint.sh > /dev/null 2>&1 & sleep 2
```

Comment:
Maybe better do an appropriate wait loop, e.g. `for i in $(seq 1 10); do docker info -f '.' && break; sleep .5; false; done`. Note that the `false` makes sure we error out if docker is not alive.

However, why can't we just set this as the ENTRYPOINT in our tests-runner image? Or specify it explicitly with

```
image:
  name: $CI_REGISTRY_IMAGE/tests-runner
  entrypoint: dockerd-entrypoint.sh
```

Reply:
We cannot use the entrypoint as-is because gitlab-ci needs the entrypoint to either be empty, or to be a shell. For reference, you can see _part_ of what gitlab-ci is doing with the entrypoint here: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/378489864
In a few words: it first exec the entrypoint, then gives it a script that, among other things, tries to exec another shell. After that (not visible in this log), the content of the "script" part of `.gitlab-ci.yml` is passed through stdin (along with a bunch of added commands and variables; `set -eo pipefail` is one of them).

I created a new entrypoint that uses the loop you describe, then runs the gitlab-ci script at the end (or fail if dockerd could not be started).